src/haddock/modules/sampling/rigidbody/__init__.py

```diff
@@ -98,48 +206,23 @@ def _run(self) -> None:
         else:
             ambig_fnames = None
 
-        # Prepare the jobs
-        idx = 1
+        start = datetime.now()
         self.output_models: list[PDBFile] = []
         self.log("Preparing jobs...")
-        for combination in models_to_dock:
-            for _i in range(sampling_factor):
-                # assign ambig_fname
-                if ambig_fnames:
-                    ambig_fname = ambig_fnames[idx - 1]
-                else:
-                    ambig_fname = self.params["ambig_fname"]
-
-                seed = self.params["iniseed"] * idx
-                # prepare cns input
-                rigidbody_input = prepare_cns_input(
-                    idx,
-                    combination,
-                    self.path,
-                    self.recipe_str,
-                    self.params,
-                    "rigidbody",
-                    ambig_fname=ambig_fname,
-                    default_params_path=self.toppar_path,
-                    native_segid=True,
-                    less_io=self.params["less_io"],
-                    seed=seed,
-                )
-
-                log_fname = f"rigidbody_{idx}.out"
-                output_pdb_fname = f"rigidbody_{idx}.pdb"
-
-                # Create a model for the expected output
-                model = PDBFile(output_pdb_fname, path=".", restr_fname=ambig_fname)
-                model.topology = [e.topology for e in combination]  # type: ignore
-                model.seed = seed
+        if self.params["mode"] == "batch":
+            cns_input = self.prepare_cns_input_sequential(
+                models_to_dock, sampling_factor, ambig_fnames  # type: ignore
+            )
 
-                self.output_models.append(model)
+        else:
+            cns_input = self.prepare_cns_input_parallel(
```

Comment:
Because you know that the parallel approach is better for high number of `models_to_dock * sampling_factor`.
Maybe checking that `models_to_dock * sampling_factor < 3000 or self.params["mode] == "batch"` to trigger the `prepare_cns_input_sequential` else  `prepare_cns_input_parallel` ?

Reply:
yep we can do that, but needs to come up with a magic number. Since the speedup threshold will depend on the number of cores and processor architecthure - I'd leave as it.